### PR TITLE
Add a guide for creating own twig functions with new example

### PR DESCRIPTION
**Pull request by Tschucki**
Referring to [PR 462](https://github.com/shopware/docs/pull/462)

I added a new example with md5.

### CODE REVIEW

**Comment by Tschucki** [comment]
Bump.

**Review by ssltg — APPROVED** [review]
Thanks for your contribution and patience (sometimes things get lost in the daily business, sorry for that).

**Comment by ssltg** [comment]
@Tschucki Maybe you have to push again to rerun the checks.

**Review by tobiasberge — APPROVED** [review]

File: guides/plugins/plugins/storefront/add-custom-twig-function.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+Imagine you want to call a PHP script from the Twig template during the theme development to create a MD5-Hash for example. For these cases we can create our own twig functions.
+
+In our example we will pass a string to the Twig Function and return a MD5-Hash.
+
+{% hint style="info" %}It is not recommended to use twig functions in order to retrieve data from the database. In these cases a DataResolver could come in handy.{% endhint %}
+
+## Prerequisites
+
+In order to create your own Twig Function for your plugin, you first need a plugin as base. Therefore, you can refer to the [Plugin Base Guide](../plugin-base-guide.md).
```
Comment (sushmangupta) [inline guides/plugins/plugins/storefront/add-custom-twig-function.md:13]:
```suggestion
In order to create your own twig function for your plugin, you first need a plugin as base. Therefore, you can refer to the [Plugin Base Guide](../plugin-base-guide.md).
```

File: guides/plugins/plugins/storefront/add-custom-twig-function.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
+# Add custom Twig Functions
```
Comment (sushmangupta) [inline guides/plugins/plugins/storefront/add-custom-twig-function.md:1]:
```suggestion
# Add Custom Twig Functions
```

File: guides/plugins/plugins/storefront/add-custom-twig-function.md [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+In our example we will pass a string to the Twig Function and return a MD5-Hash.
+
+{% hint style="info" %}It is not recommended to use twig functions in order to retrieve data from the database. In these cases a DataResolver could come in handy.{% endhint %}
+
+## Prerequisites
+
+In order to create your own Twig Function for your plugin, you first need a plugin as base. Therefore, you can refer to the [Plugin Base Guide](../plugin-base-guide.md).
+
+## Creating Twig Function
```
Comment (sushmangupta) [inline guides/plugins/plugins/storefront/add-custom-twig-function.md:15]:
```suggestion
## Creating twig function
```

File: guides/plugins/plugins/storefront/add-custom-twig-function.md [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+
+In the following sections we will create and expand all necessary files for the Twig function to work. These are actually exactly 2 files. First the PHP file with the Twig Functions itself and the services.xml.
```
Comment (sushmangupta) [inline guides/plugins/plugins/storefront/add-custom-twig-function.md:17]:
```suggestion
In the following sections, we will create and expand all necessary files for the twig function to work. There are two such files:
* PHP file with the twig functions itself and
* Services.xml
```

File: guides/plugins/plugins/storefront/add-custom-twig-function.md [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+
+### Creating the Twig Function
```
Comment (sushmangupta) [inline guides/plugins/plugins/storefront/add-custom-twig-function.md:19]:
```suggestion
### Creating the twig function
```

File: guides/plugins/plugins/storefront/add-custom-twig-function.md [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+
+For the sake of clarity, we will create a folder named Twig within the src folder. Within this we create a new php file. You can name this file whatever you want.
+
+Let's have a look at the created file.
```
Comment (sushmangupta) [inline guides/plugins/plugins/storefront/add-custom-twig-function.md:23]:
```suggestion
For clarity, create a folder named `Twig` within the `src` folder. Then create a new php file with desired file name within the `Twig` folder. Refer to the below example :
```

File: guides/plugins/plugins/storefront/add-custom-twig-function.md [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+    {
+        return md5($str);
+    }
+}
+```
+{% endcode %}
+
+Of course you can do everything in the createMd5Hash function that PHP can do.
+
+What is still missing is the registration of the service in the DI container.
```
Comment (sushmangupta) [inline guides/plugins/plugins/storefront/add-custom-twig-function.md:54]:
```suggestion
Of course, you can do everything in the `createMd5Hash` function that PHP can do, but the service.xml handles registration of the service in the DI container.
```

File: guides/plugins/plugins/storefront/add-custom-twig-function.md [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+    <services>
+        <service id="SwagBasicExample\Twig\SwagCreateMd5Hash" public="true">
+            <tag name="twig.extension"/> <!--Required-->
+        </service>
+    </services>
+...
+```
+{% endcode %}
+
+After all that is done, you can access this Twig function within your plugin.
```
Comment (sushmangupta) [inline guides/plugins/plugins/storefront/add-custom-twig-function.md:68]:
```suggestion
Once done, you can access this `TwigFunction` within your plugin.
```

File: guides/plugins/plugins/storefront/add-custom-twig-function.md [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+
+### Use the Twig function in your template
```
Comment (sushmangupta) [inline guides/plugins/plugins/storefront/add-custom-twig-function.md:70]:
```suggestion
### Use twig function in template
```

**Review by sushmangupta — CHANGES_REQUESTED** [review]

**Comment by sushmangupta** [comment]
@Tschucki , apologies for the delayed response. 
I have added a few suggestions. Can you please take a look at it possibly?

**Review by sushmangupta — APPROVED** [review]

**Comment by sushmangupta** [comment]
Sorry for the delayed response and thanks for contributing @Tschucki :)

**Pull request merged**